langchain/tools/powerbi/tool.py

```diff
@@ -23,110 +23,135 @@ class QueryPowerBITool(BaseTool):
 
     name = "query_powerbi"
     description = """
-    Input to this tool is a detailed and correct DAX query, output is a result from the dataset.
-    If the query is not correct, an error message will be returned.
-    If an error is returned with Bad request in it, rewrite the query and try again.
-    If an error is returned with Unauthorized in it, do not try again, but tell the user to change their authentication.
+    Input to this tool is a detailed question about the dataset, output is a result from the dataset. It will try to answer the question using the dataset, and if it cannot, it will ask for clarification.
 
-    Example Input: "EVALUATE ROW("count", COUNTROWS(table1))"
+    Example Input: "How many rows are in table1?"
     """  # noqa: E501
+    llm_chain: LLMChain
     powerbi: PowerBIDataset = Field(exclude=True)
+    template: Optional[str] = QUESTION_TO_QUERY
+    examples: Optional[str] = DEFAULT_FEWSHOT_EXAMPLES
     session_cache: Dict[str, Any] = Field(default_factory=dict, exclude=True)
+    max_iterations: int = 5
 
     class Config:
         """Configuration for this pydantic object."""
 
         arbitrary_types_allowed = True
 
+    @validator("llm_chain")
+    def validate_llm_chain_input_variables(  # pylint: disable=E0213
+        cls, llm_chain: LLMChain
+    ) -> LLMChain:
+        """Make sure the LLM chain has the correct input variables."""
+        if llm_chain.prompt.input_variables != [
+            "tool_input",
+            "tables",
+            "schemas",
+            "examples",
+        ]:
+            raise ValueError(
+                "LLM chain for InputToQueryTool must have input variables ['tool_input', 'tables', 'schemas', 'examples']"  # noqa: C0301 E501 # pylint: disable=C0301
```

Comment:
would be nice to tell the user what variables it actually found

Reply:
good one, added